doc.go

```diff
@@ -271,5 +297,141 @@ from the closest provider.
 Providers that have unmet dependencies will be eliminated from the chain
 unless they're Required.
 
+Best practices
+
+The remainder of this document consists of suggestions for how to use nject.
+
+Contributions to this section would be welcome.  Also links to blogs or other
+discussions of using nject in practice.
+
+For tests
+
+The best practice for using nject inside a large project is to have a few
+common chains that everyone imports.
+
+Most of the time, these common chains will be early in the sequence of
+providers.  Customization of the import chains happens in many places.
+
+This is true for services, libraries, and tests.
+
+For tests, a wrapper that includes the stanard chain makes it eaiser
+to write tests.
+
+	var CommonChain = nject.Sequence("common",
+		context.Background,
+		log.Default,
+		things,
+		used,
+		in,
+		this,
+		project,
+	)
+
+	func RunTest(t *testing.T, testInjectors ...any) {
+		err := nject.Run("RunTest",
+			t,
+			CommonChain,
+			nject.Sequence(t.Name(), testInjectors...))
+		assert.NoError(t, err, nject.DetailedError(err))
+	}
+
+	func TestSomething(t *testing.T) {
+		t.RunTest(t, Extra, Things, func(
+			ctx context.Context,
+			log *log.Logger,
+			etc Etcetera ) {
+			assert.NotNil(t, ctx)
+		})
+	}
+
+Displaying errors
+
+If nject cannot bind or run a chain, it will return error.  The returned
+error is generally very good, but it does not contain the full debugging
+output.
+
+The full debugging output can be obtained with the DetailedError function.
+
+	err := nject.Run("some chain", some, injectors)
+	if err != nil {
+		if details := nject.DetailedError(err); details != err.Error() {
+			log.Println("Detailed error", details)
+		}
+		log.Fatal(err)
+	}
+
+Reorder
+
+The Reorder() decorator allows injection chains to be fully or partially reordered.
+The recommended usage is to use Reorder() sparingly, but use it.
+
+Reorder solves a common problem: providing optional options to an injected dependency.
+
+	var ThingChain = nject.Sequence("thingChain",
+		nject.Shun(DefaultThingOptions),
+		ThingProvider,
+	}
+
+	func OverrideThingOptions(options ...ThingOption) nject.Provider {
+		return nject.Reorder(func() []ThingOption) {
+			return options
+		}
+	}
+
+	func DefaultThingOptions() []ThingOption {
+		return []ThingOption{
+			StanardThingOption
+		}
+	}
+
+	func ThingProvider(options []ThingOption) *Thing {
+		return thing.Make(options...)
+	}
+
+Because the default options are marked as Shun, they'll only be included
+if they have to be included.  If a user of thingChain wants to override
+the options, they simply need to include mark their override as Reorder.
+
+	nject.Run("run",
+		ThingChain,
+		OverrideThingOptions(thing.Option1, thing.Option2),
+	)
+
+Self-cleaning
+
+Recommened best practices to have injectors shutdown the things they themselves start. They
+should do their own cleanup.
+
+Inside tests, that's easy because an injector in a test can use t.Cleanup().
```

Comment:
```suggestion
Inside tests, an injector can use t.Cleanup() for this.
```

Generally want to avoid words like "easy". This sentence might need more fluff than how I've left it though.